### PR TITLE
chore: add required fields to the Provider spec (fix #96)

### PR DESCRIPTION
**Pull request by vbelouso**
## Description

Fix for #96 
Mark `defaultModel`  and `defaultProvider` as required fields.

## Type of change

- [ ] Refactor
- [ ] New feature
- [x] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: api/v1alpha1/olsconfig_types.go [hunk 1]
```diff
@@ -69,13 +69,13 @@ type OLSSpec struct {
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Disable Authorization",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
 	DisableAuth bool `json:"disableAuth,omitempty"`
 	// Default model for usage
-	// +kubebuilder:validation:Required
-	// +required
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Default Model",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced"}
-	DefaultModel string `json:"defaultModel"`
+	DefaultModel string `json:"defaultModel,omitempty"`
```
Comment (bparees) [inline api/v1alpha1/olsconfig_types.go:73]: default model should also be required

**Comment by raptorsun** [comment]
> Changes look good . See new block in the generated operator.yaml - Will unhold/approve once I verify where its coming from.

The new format in operator.yaml should come from the update on controller runtime package https://github.com/openshift/lightspeed-operator/pull/92

**Comment by vbelouso** [comment]
> > Changes look good . See new block in the generated operator.yaml - Will unhold/approve once I verify where its coming from.
> 
> The new format in operator.yaml should come from the update on controller runtime package #92

It came from an updated version of OPM
I have version `1.39.0` installed, but in Makefile we use version `1.23`

Xavi will test the work with the new version. 
If something goes wrong, I will roll back my version to `1.23`

More info 
https://olm.operatorframework.io/docs/reference/file-based-catalogs/#olmbundleobject-deprecated
https://olm.operatorframework.io/docs/reference/file-based-catalogs/#olmcsvmetadata

**Comment by raptorsun** [comment]
> > > Changes look good . See new block in the generated operator.yaml - Will unhold/approve once I verify where its coming from.
> > 
> > 
> > The new format in operator.yaml should come from the update on controller runtime package #92
> 
> It came from an updated version of OPM I have version `1.39.0` installed, but in the Makefile we use version `1.23`
> 
> Xavi will test the new version. If something goes wrong, I will roll back my version to `1.23.`
> 
> More info https://olm.operatorframework.io/docs/reference/file-based-catalogs/#olmbundleobject-deprecated https://olm.operatorframework.io/docs/reference/file-based-catalogs/#olmcsvmetadata

nice, if the new version works well, let's use the newer OPM :)

**Comment by bparees** [comment]
@xrajesh this PR is still held, did you also intend to unhold it?

**Pull request merged**